src/Framework/TestSuite.php

```diff
@@ -107,12 +107,12 @@ class TestSuite implements IteratorAggregate, Reorderable, SelfDescribing, Test
     protected $foundClasses = [];
 
     /**
-     * @var null|array<ExecutionOrderDependency>
+     * @var null|list<ExecutionOrderDependency>
```

Comment:
if you were to default this property to an empty array; would it be possible to remove the null and thus have a cleaner type?

Reply:
Yes, at the cost of having another "need to (re)build dependency tree" flag. Currently there is no direct candidate for that, eventhough I could reuse the existing `count === -1` mechanism.

Reply:
Ah, the null signifies that the dependency tree needs to be rebuilt; check.

Reply:
p.s. this piece of info could be added as documentation to the DocBlock for future readers :)